### PR TITLE
Ironic: add option to disable autoclean + conf update

### PR DESCRIPTION
**Pull request by skazi0**
A new attribute was added to disable automatic cleaning of baremetal nodes. This is especially useful during testing and development as the cleaning procedure can take a lot of time. For production it is not recommended but there are also some cases where it can be useful (e.g. when one tenant wants to quickly re-cycle hardware).

Second commit is update of nova conf to remove some deprecation warnings.

### CODE REVIEW

File: chef/cookbooks/nova/templates/default/nova.conf.erb [hunk 1]
```diff
@@ -145,11 +145,13 @@ protocol = <%= @glance_server_protocol %>
...
-admin_password = <%= @ironic_settings[:service_password] %>
-admin_url = <%= @keystone_settings['protocol'] %>://<%= @keystone_settings['internal_url_host'] %>:<%= @keystone_settings['admin_port'] %>/<%= @ironic_settings[:keystone_version] %>/
-admin_tenant_name = <%= @keystone_settings['service_tenant'] %>
+auth_type = password
+auth_url = <%= @keystone_settings['internal_auth_url'] %>
+username = <%= @ironic_settings[:service_user] %>
+password = <%= @ironic_settings[:service_password] %>
+project_name = <%= @keystone_settings['service_tenant'] %>
+project_domain_name = <%= @keystone_settings['admin_domain']%>
+user_domain_name = <%= @keystone_settings['admin_domain'] %>
```
Comment (jsuchome) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: Are those last three items really needed? They are not specific to ironic and we're already saving them in keystone section...

(I do not know about ironic, so I'm just curious.. in general the second commit could be better with better message)

**Comment by jsuchome** [comment]
First commit looks good, I think second commit should be explained better.

File: chef/cookbooks/nova/templates/default/nova.conf.erb (hunk 1 above)
Comment (skazi0) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: 1. They are needed. I tried without them first and it failed to authenticate.
2. This seems to be a new recommendation from upstream to define auth parameters in each section separately with a standardized naming. I think this is due to migration from `keystoneclient` to `keystoneauth` library. See "Configure credentials for accessing other OpenStack services." under https://docs.openstack.org/project-install-guide/baremetal/draft/install-obs.html#configuring-ironic-conductor-service).

Comment (skazi0) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: Relevant reno: https://github.com/openstack/nova/blob/master/releasenotes/notes/deprecate-old-auth-parameters-948d70045335b312.yaml

**Pull request merged**